lib/common/variadic/uint32OrHash.go

```diff
@@ -16,6 +17,18 @@ type Uint32OrHash struct {
 	value interface{}
 }
 
+func FromHash(hash common.Hash) *Uint32OrHash {
```

Comment:
Nit/I think this should either be turned into a receiver to make the behavior more clear. I would say just rename but `NewUint32OrHashFromHash` isn't the best name, but if you have a good name I am open to this idea.

I say this because using this from another package will be something like `variadic.FromHash` which I don't think is very intuitive. Perhaps could also make this its own package to fix this? I am open to solutions. This also applies to the function below

Reply:
I see, I also think that this variadic package can be replaced by a new type in the `types` package called `BlockKey` which basically can be the block number (a numeric type) or the block hash. Then BlockKey can be a generic struct over these two options (hash or integer) and then we we should do the encoding accordingly to the type and then replace this variadic package. I remember to see something similar in the @timwu20  blockchainDB PR, maybe we can use it, until then I will place a TODO here wdyt?

Reply:
But will create a func that constraint the argument to `common.Hash` or `~int | ~uint | ~uint32` then we don't need to use `MustNewUint32OrHash` where we can panic at runtime bc of a wrong arg type

```go 
func Uint32OrHashFrom[T common.Hash | ~int | ~uint | ~uint32](value T) *Uint32OrHash {
	return &Uint32OrHash{
		value: value,
	}
}

```